Nomad Docs: Enhance client intro release note & add usage guide (#1384)
hashicorp/web-unified-docs: content/nomad/v1.11.x/content/docs/release-notes/nomad/v1-11-x.mdx

Nomad's client node identity feature uniquely identities each Nomad client node
and provides an authentication mechanism for nodes to make RPC calls to the
Nomad servers. 

Introduce Nomad clients to the cluster with JWT tokens. Configure Nomad servers
with introduction enforcement levels that dictate how clients join the cluster.
This approach results in logs and metrics to detail introduction violations.
Once registered, Nomad clients are now provided with an identity token, used for
RPC communication which is periodically renewed.

The client node introduction and identity feature is like multi-factor
authentication for your Nomad clusters. It does not replace mTLS but adds
a second layer of security to prevent an unauthorized client from joining a

> Nomad cluster. 

Reviewer: ```suggestion
The client node introduction and identity feature functions as multi-factor
authentication for your Nomad clusters. It does not replace mTLS. Instead, 
it adds a second layer of security to prevent an unauthorized client from 
joining a Nomad cluster. 
```
The "like" is holding a lot of weight in this sentence. My suggestions here, including breaking the statement up into three sentences, is meant to help make the simile easier to digest.

Each layer answers a distinct question. 

- Networking: Can the client reach the server?
- mTLS: Does the client have valid certificates for the cluster?
- Client introduction token: Does the client have a valid token to join the
cluster?

> Using a client introduction token gives you the added benefit of additional control over misconfigured clients trying to join the cluster. You can specify node names, node pools, and TTLs for the tokens you generate.

Reviewer: ```suggestion
Using a client introduction token gives you additional control over misconfigured clients trying to join the Nomad cluster because you can specify node names, node pools, and TTLs for the tokens you generate.
```
This makes more sense as a single sentence to me (benefit + reason)
